core/src/main/java/com/github/mizool/core/password/PasswordHasherRegistry.java

```diff
@@ -31,8 +37,27 @@
 
     public PasswordHasher getHasher(String algorithm)
     {
-        return Iterables.getOnlyElement(
-            Iterables.filter(
-                passwordHashers, passwordHasher -> passwordHasher.isResponsibleFor(algorithm)));
+        return Iterables.getOnlyElement(Iterables.filter(passwordHashers,
+            passwordHasher -> passwordHasher.isResponsibleFor(algorithm)));
+    }
+
+    public static PasswordHasher getDefaultHasher()
+    {
+        Iterable<PasswordHasher> hasherInstances = MetaInfServices.instances(PasswordHasher.class);
```

Comment:
Don't interact with `MetaInfServices`, use the injected `Instance<PasswordHasher> passwordHashers` to stay compatible with CDI-based applications. For non-CDI environments like `PasswordHashTool`, let's use an MetaInfServices-to-Instance adapter. (I don't know offhand if we already have one in this code base, contact me if not and I'll look it up elsewhere.)